decentraland/components/presentational/BuyLand.js

```diff
@@ -1,38 +1,60 @@
 import React from "react";
-import { StyleSheet, View } from "react-native";
+import { StyleSheet, View, Text } from "react-native";
 import { responsiveHeight } from "react-native-responsive-dimensions";
 import PropTypes from "prop-types";
 import LandForSale from "./LandForSale";
 import Button from "./Button";
 import Colors from "@constants/Colors";
 
 export default function BuyLand(props) {
-  const { landForSale, onBuy, waitingForAccountSetup } = props;
+  const { landForSale, onBuy, accountInfo } = props;
+  const {
+    account,
+    setupInProgress: waitingForAccountSetup,
+    fundedWithEthers,
+    fundedWithMana,
+    approvedMarketplace,
+  } = accountInfo;
+
+  let waitingMessage = "";
+  if (waitingForAccountSetup) {
+    if (account == null) waitingMessage = "Generating account...";
+    else if (!fundedWithEthers) waitingMessage = "Funding account with ETH...";
+    else if (!fundedWithMana && !approvedMarketplace)
+      waitingMessage = "Funding account with MANA and approving Marketplace...";
```

Comment:
```suggestion
      waitingMessage = "Funding account with MANA and approving marketplace...";
```

Reply:
The user has no reason to know that "marketplace" is the name of the contracts, so it's more clear as a general concept if it's not capitalized